refactor(Talk): altera o campo speaker para speaker_id e insere a chave estrangeira na classe Talk

rafaelvnunes opened the pull request:
Verificar se o `AdminListCreateTalksView.create` está validando a existência do Speaker da maneira correta (não havia no saphira um caso parecido). Assumimos que o frontend envia um Speaker com todos os atributos, ao invés de apenas o id, se for apenas o id precisamos alterar o código.

Fecha a tarefa #66.

rafaelvnunes committed:
feat(models.py): altera campo de texto para uma relação com a classe Speaker em Talk

A classe Speaker precisou ser movida para antes do Talk para ela conseguir ser referenciada/utilizada em Talk.

Co-authored-by: Antonio Espindula <antonio.pedro.espindula@gmail.com>

rafaelvnunes committed:
feat(migrations): cria o arquivo de migração alterando a classe Talk

Agora palestrante deixou de ser um campo de texto com o nome do palestrante para ser uma chave estrangeira que relaciona a entidade Palestra (Talk) com a entidade Palestrante (Speaker)

Co-authored-by: Antonio Espindula <antonio.pedro.espindula@gmail.com>

rafaelvnunes committed:
feat(serializers.py): remove o atributo speaker do TalkSerializer

Agora a Talk tem um relacionamento com Speaker, ao invés de um campo de texto para o nome do palestrante

Co-authored-by: Antonio Espindula <antonio.pedro.espindula@gmail.com>

rafaelvnunes committed:
feat(views.py): valida se o objeto Speaker existe antes de criar um Talk

Como o frontend envia um objeto Speaker para o backend salvar na palestra é necessário verificar se o Speaker existe antes de criar o Talk

Co-authored-by: Antonio Espindula <antonio.pedro.espindula@gmail.com>

rafaelvnunes commented:
Inclusive, esse erro do CI é justamente porque já existem dados no banco da COSSI que os palestrantes eram apenas campos de texto aí para os dados antigos ele não sabe o que fazer com a chave estrangeira. Temos que pensar em uma estratégia para esses dados antigos.

mercuryVM reviewed (changes requested):
Boa implementação, os modelos imagino que estão corretos.

No entanto, ainda há necessidade de alterar os testes unitários que utilizavam as views alteradas (só ver que os testes do github falharam) e alterar para o novo modelo de entrada

caio-bernardo commented:
Tirando a necessidade de arrumar os testes unitários, o código ta perfeito. Sobre o conflito com dados antigos, pode ficar tranquilo. A base de dados em produção desse ano só vai ser criada quando o saphira estiver pronto. Então não vai ter conflito com dados antigos. Resolvendo os testes eu aprovo a PR. Seis tão mandando bem demais!

caio-bernardo reviewed (changes requested):
- [x] atualizar os testes para usarem o modelo Speaker

rafaelvnunes committed:
test(Talk): atualiza os testes que envolvem o modelo Talk para incluir Speaker

O atributo speaker do modelo Talk deixou de ter o campo de texto speaker e agora tem uma chave estrangeira com o uuid de Speaker

rafaelvnunes commented:
Atualizei os testes e fiz um git rebase para corrigir um erro que não tinha notado antes em um commit anterior
